### PR TITLE
use tracing crate instead of a file based logger

### LINKED ISSUE #17: Replace the file based logger
Right now the project uses a file-based logger. 
Consider using the [tracing](https://crates.io/crates/tracing) crate and writing to standard error instead of writing to a file.
You also need to remove the configuration parameter for the logger.
Check that it can log using async.

### PR DESCRIPTION
**Pull request by a-moreira**
closes https://github.com/lambdaclass/libtorrent-rs/issues/17

### CODE REVIEW

File: dtorrent/src/main.rs [hunk 1]
```diff
@@ -19,13 +19,10 @@ fn main() {
...
-    let logger = Logger::new(&config.log_directory, config.max_log_file_kb_size)
-        .expect("logger could not be created");
     let client_peer_id = "client_peer_id".to_string();
     let mut server = BtServer::new(
         torrent_with_status,
         config,
-        logger.new_sender(),
         client_peer_id,
     );
     println!("Initializing server ...");
```
Comment (edg-l) [inline dtorrent/src/main.rs]: should use info!

File: dtorrent/src/peer/peer_session.rs [hunk 2]
```diff
@@ -319,8 +310,8 @@ impl PeerSession {
         self.check_last_piece_block(piece_index, entire_blocks_in_piece, stream)?;
 
         self.validate_piece(&self.piece, piece_index)?;
-        self.logger_sender
-            .info(&format!("Piece {} downloaded!", piece_index));
+
+        info!("Piece {} downloaded!", piece_index);
 
         let remaining_pieces = self.torrent_status.downloaded_pieces();
         println!(
```
Comment (edg-l) [inline dtorrent/src/peer/peer_session.rs]: should use info!

File: dtracker/src/http_server/server.rs [hunk 3]
```diff
@@ -47,21 +44,20 @@ impl Server {
     /// Handles new connections to the server
     pub fn serve(&self) -> std::io::Result<()> {
         let started_msg = format!("Serving on http://0.0.0.0:{}", self.port);
-        self.logger_sender.info(&started_msg);
+        info!(started_msg);
         println!("{}", started_msg);
```
Comment (edg-l) [inline dtracker/src/http_server/server.rs]: should use info!

**Review by edg-l — CHANGES_REQUESTED** [review]
Looks good, but most prints in the library code should be turned into either info! or debug! calls.

**Comment by a-moreira** [comment]
@edg-l done

edit: sorry, removed the other reviewers unwillingly 😅

File: dtracker/src/http_server/server.rs [hunk 4]
```diff
@@ -45,7 +44,7 @@ impl Server {
     pub fn serve(&self) -> std::io::Result<()> {
         let started_msg = format!("Serving on http://0.0.0.0:{}", self.port);
         info!(started_msg);
-        println!("{}", started_msg);
+        info!("{}", started_msg);
```
Comment (edg-l) [inline dtracker/src/http_server/server.rs]: These 3 lines can be combined into 1 (the last 2 are repeated).

The `info!` macro is at the same time a formatter and a print call.

So u can just put the `format!` parameters into a single `info!` call.

**Review by edg-l — CHANGES_REQUESTED** [review]

File: dtracker/src/http_server/server.rs (hunk 4 above)
Comment (a-moreira) [reply dtracker/src/http_server/server.rs]: yep, you are right, went too fast through this one

Comment (a-moreira) [reply dtracker/src/http_server/server.rs]: I also did a `cargo fmt --all`

**Comment by a-moreira** [comment]
should I remove the `logger` crate?

**Comment by edg-l** [comment]
I would if its unused then yes.

**Review by edg-l — APPROVED** [review]
lgtm

**Review by Oppen — APPROVED** [review]

**Pull request merged**